src/frontend/js/toggle-dashboard-section.ts

```diff
@@ -18,3 +20,97 @@ async function toggleDashboardSection({ target }: Event): Promise<void> {
+    };
+    
+    feedLoading.classList.remove("hidden");
+    reloadButton.classList.add("hidden");
+    const url = feedWidget.attributes[1].value;
+
+    // set timeout
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => {
+      controller.abort();
+      handleError();
+    }, 5000);
+
+    fetch(url, {signal: controller.signal}).then(async (response: Response) => {
+      if (response.status === 200 && response.ok) {
+        feedLoading.classList.add("hidden");
+        clearTimeout(timeoutId);
+        let parser = new DOMParser;
+        let feedHtml = parser.parseFromString(
+          parser.parseFromString(await response.text(), "text/xml").documentElement.innerHTML, "text/html");
+
+        let firstEntry = 0, lastEntry = 3; 
+        feedWidget.innerHTML += await loadRssEntries(feedHtml, firstEntry, lastEntry);
+        const readMoreButton = document.getElementById("read-more")
+        readMoreButton.classList.remove("hidden");
+        readMoreButton.addEventListener("click", async () => {
+          
+          firstEntry = lastEntry;
+          lastEntry += 3;
+
+          feedWidget.innerHTML += await loadRssEntries(feedHtml, firstEntry, lastEntry);
+
+          if (feedHtml.querySelectorAll("item").length - 1 <= firstEntry) {
+            readMoreButton.classList.add("hidden");
+
+            let entries = document.querySelectorAll(".rss-entry");
+            entries[entries.length - 1].classList.remove("border-b")
+            entries[entries.length - 1].classList.remove("border-gray-400")
+            entries[entries.length - 1].classList.remove("mb-2");
+          }     
+        }); 
+      } else {
+        throw Error("NetworkError. " + url + "does not respond.");
+      }
+    }).catch ((err) => {
+      // console.error(err);
+      handleError();
+    });
+  }
+}
+
+
+async function loadRssEntries(doc: Document, firstEntry: number, lastEntry: number) : Promise<String> {
+  let feed = "";
+
+  Array.from(doc.querySelectorAll("item")).slice(firstEntry, lastEntry).forEach((node) => {
+    let title: String, link: String, content: String;
+
+    // get title, link and content this way as they don't have a consistent index in the HTMLCollection
+    Array.from(node.children).forEach(element => {
+      if (element.localName === "title") title = element.textContent;
+      if (element.localName === "link") link = element.nextSibling.nodeValue;
+      if (element.localName === "content:encoded") content = element.textContent;
+    });
+
+    if (title && link && content) {
+      let rssEntry = '<div class="rss-entry pb-2 border-b border-gray-400 mb-2">'
+        + '<a href=' 
+        + link 
+        + 'target="_blank" rel="noopener noreferrer" class="block py-b font-bold text-blue-500 hover:text-blue-700">'
+        + title 
+        + '</a>' + '<div class="text-gray-600 py-2"></div>'
+        + '<p class="pb-1">' + content.split(" ").slice(0, 25).join(" ") + ' . . . </p>'
```

Comment:
```suggestion
        + '<p class="pb-1">' + content + '</p>'
```

Reply:
But this one would always render the full article to the news feed, no matter how long it is, wouldn't it? Wouldn't it make sense to limit it to snippet, so that interested readers just check the link and get the full and well formatted article in a new window as it was in the old feed?